File: Programmers/codingTest_Kit/Heap/42626.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> scoville, int K) {
    priority_queue<int, vector<int>, greater<int>> PQ;
    for(int i = 0; i < scoville.size(); i++) PQ.push(scoville[i]); 
    int count = 0;
    while(PQ.size() > 1 && PQ.top() < K){
        int first = PQ.top();
        PQ.pop();
        int second = PQ.top();
        PQ.pop();
        int result = first + (second * 2);
        PQ.push(result);
        count++;
    }
    if(PQ.top() < K) count = -1;
    return count;
}

File: Programmers/codingTest_Kit/Heap/42626.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (int i = 0; i < scoville.length; i++) {
            pq.add(scoville[i]);
        }

        while (pq.size() > 1 && pq.peek() < K) {
            int first = pq.poll();
            int second = pq.poll();

            pq.add(first + 2 * second);
            answer++;
        }

        if (pq.size() <= 1 && pq.peek() < K) {
            answer = -1;
        }

        return answer;
    }
}